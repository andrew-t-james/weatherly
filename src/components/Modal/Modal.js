import React from 'react';
import Search from '../Search/Search.js';

const Modal = (props) => {
  const { updateLocation, hasError, hasLocation } = props;

  return (
    <div>
      <Search
        updateLocation={updateLocation} />
      {!hasLocation && !hasError && <h2>Welcome To Weatherly</h2>}
      {hasError && <h2>Please Enter a valid City State or Zip</h2> }
    </div>
  );
};

export default Modal;