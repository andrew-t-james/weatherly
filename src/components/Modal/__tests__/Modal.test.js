import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Modal from '../Modal.js';

describe('App unit tests', () => {
  test('it should render a welcome message if noLocation present', () => {
    const expected = 'Welcome To Weatherly';
    const modal = shallow(<Modal hasLocation={false} />);
    const actual = modal.find('h2').text();

    expect(actual).toBe(expected);
  });

  test('it should render an err message if hasError present', () => {
    const expected = 'Please Enter a valid City State or Zip';
    const modal = shallow(<Modal hasError={true} />);
    const actual = modal.find('h2').text();

    expect(actual).toBe(expected);
  });

  test('it should match snapshot', () => {
    const modal = renderer
      .create(<Modal />)
      .toJSON();

    expect(modal).toMatchSnapshot();
  });
});