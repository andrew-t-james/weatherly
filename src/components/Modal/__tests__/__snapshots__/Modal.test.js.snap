// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App unit tests it should match snapshot 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <h1
      className="header__title"
    >
      Weathrly
    </h1>
    <form
      className="search-form"
      onSubmit={[Function]}
    >
      <input
        className="search-form__input"
        name="location"
        onChange={[Function]}
        placeholder="Search by City and State or Zip"
        type="text"
        value=""
      />
      <button
        className="search-form__button"
        disabled={true}
        type="submit"
      >
        Search
      </button>
    </form>
  </header>
  <div
    className="welcome"
  >
    <h2
      className="welcome__title"
    >
      Welcome To Weathrly
    </h2>
    <video
      autoPlay={true}
      className="welcome__video"
      loop={true}
      muted={true}
      poster="storm-poster.png"
    >
      <source
        src="storm.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;
